### ISSUE TITLE
Split website for current/ongoing STAT 545 offering and STAT545.com

### ISSUE THREAD

**Issue by jennybc** [STAT545-UBC/STAT545-UBC-original-website#71]
This is meant to start a discussion.

I am beginning to contemplate a split between the website for the "current run" of STAT 545 and the (existing) course content.

Thoughts rattling around in my head re: why and how:

  * The current course should fall back to having its own site at STAT545-UBC.github.io (or elsewhere, whatever the instructor wants). And STAT545.com could lose all of the logistical and date-specific stuff that comes from specific course runs, i.e. it can become content-only and not mention specific homework assignment, due dates, etc.
  * I am no longer the "owner" of STAT 545 at UBC and obviously that content needs to keep evolving. I do feel like a major "owner" of the content currently served at STAT545.com and, literally, I own the domain. This will become increasingly awkward, because either I need to keep the course content sort of frozen (obviously a bad idea) or take an active role in evolution of the content (not practical).
  * We could do this in phases or on an as-needed, as-useful basis. Step 1 might be just to separate perishable, time-sensitive stuff from more durable bits of curriculum. But maybe we get through 2018/2019 with joint management? It kind of depends on everyone's appetite for or resistance to change.
  * The STAT 545 website is made using fairly antiquated methods (an Rmd website, manual rendering of files) and I'm starting to think about migrating it to something more modern. But it makes sense to sort out this other process first.

I look forward to hearing from other current stakeholders, especially those slated to teach this fall!

**Comment by vincenzocoia** [comment]
@jennybc: I'll be instructing the class meetings wholly this fall, and agree with your overall sentiment to separate the "current run" from the content. I've been treating your content (i.e., things on the [topics notes](http://stat545.com/topics.html)) as your property, using it to underpin "guides" that I've been making for in-class use. 

I think what happens in the future largely depends on what you want to do with the content. Perhaps you want to amalgamate some/all of it into [Happy git with R](http://happygitwithr.com/) and/or other publications? And/or, we can treat STAT545.com like its own book that gets a new version each year -- and switch to a method like blogdown with Hugo.

For this year, I've created a STAT545-LMS org for learning management (where students will have their repos, submit their homework, and get feedback). I can move the homework descriptions and syllabus there, too, in which case STAT545.com would only contain content (and the class "guides" that I might continue to make).

If it makes things easier, I'm happy to chat with you about these things in-person.

**Comment by jennybc** [comment]
Also, maybe I should create a "STAT545" (no UBC) organization to eventually hold the Topics "asset" or ... plan to eventually move it back under my own account. I could still offer a web version from STAT545.com, I'm just talking about where source lives on GitHub now.

**Comment by vincenzocoia** [comment]
> OK these seem to be pretty clear Phase One "TO DO's" (agree?):

Yes, these all sound aligned to what I was thinking. I'll keep you in the loop. 

Also, I renamed the LMS org to `STAT545-UBC-classroom` -- I think it's more indicative. 

> Where do you see your class guides fitting in? 

I'll likely use the `cm**`'s for this, in the `classroom` org.

**Comment by vincenzocoia** [comment]
And, 

> Are you using that mechanism to avoid changing/adding stuff below Topics?

Yes, I wanted to leave the stuff under Topics alone. But this is getting more awkward, as the course content is evolving.

**Comment by jennybc** [comment]
> Also, I renamed the LMS org to STAT545-UBC-classroom -- I think it's more indicative.

Thinking more ... it seems like **you** should just be using the existing [STAT545-UBC](https://github.com/STAT545-UBC/) Organization, i.e. there's no need for a new Org. Or if there is, it is I who should eventually move stuff out. What's your motivation for a new Org?

I like the "classroom" name. Seems like the new site that reflects running instances of STAT 545 at UBC could be based in a repo named something like "STAT545-UBC-classroom" within the "STAT545-UBC" Org.

Re: class guides. OK, got it. If you want to keep that stuff totally separate, that is fine with me. It is conceivably easiest, because collaboration takes energy. That said, I am also not *opposed* to updates to the material in [Topics](http://stat545.com). I just can't let dialogue about that, differences in vision/goals, create a lot of time-consuming back and forth. As I'm sure is also true of you. So, feel free to make PRs for stuff that seems simple and keep doing your class guide thing when that seems easiest.

**Comment by jennybc** [comment]
I think it would be a good idea to think about having Orgs specific to each course run to hold student repos, though, because that situation is growing out of hand. Is there any talk of hosting those repos on the UBC instance of GitHub Enterprise?

It would probably be best if the STAT 545 Org only held course-level assets (your "classroom" repo, the existing Topics content, the Discussion repo).

**Comment by vincenzocoia** [comment]
> What's your motivation for a new Org?

I wanted an org for which I have ownership over, so that I could create private repos for the students.  Adding me as an owner of `STAT545-UBC` would eliminate the need for a new repo. And I like the idea of having a `classroom` repo within the already existing `STAT545-UBC` org.

Re: class guides. The small PR approach sounds reasonable to me.

**Comment by vincenzocoia** [comment]
Oops, missed your previous comment. If your experience is that it would be better to have one Org per year to handle student repos, then I'm totally on board to the two-Org situation. As for UBC's GitHub Enterprise, it's not quite ready for STAT 545. 

> It would probably be best if the STAT 545 Org only held course-level assets 

Perhaps a `students` Org that gets replaced annually, alongside the STAT 545 Org, would be most appropriate.

**Comment by jennybc** [comment]
I'm happy to do some of this once we agree on it, especially the bulk repo moving work. It always helps me find issues in various packages 😁

**Comment by vincenzocoia** [comment]
Thanks, @jennybc. 

> I'd recommend:
> 
> - Our existing STAT545-UBC Organization for course level assets. It can host your new "classroom" repo that holds time-sensitive content relevant to specific academic years. Will continue to host the main curriculum/Topics content for the foreseeable future. Also the private instructors-only repo.
> - An Organization for every future combination of STAT 545 (and 547?) * academic year. That would hold the student repos for that run.

I'm on board with this.

> I'm not sure whether to keep using a single, enduring Discussion repo or to also make that specific to each run.

I intend to use a year-specific repository to host discussions -- at least for internal matters (i.e., assignments, office hours, etc). I can encourage students to post in the public-facing Discussions repo for content matters.

> Maybe we should make these run-specific Orgs retrospectively and move stuff?

Or, a single pre-2018 repo for all of these?

**Comment by jennybc** [comment]
I can't remember the exact process (and it may have changed) but you should probably create the 2018 Org(s) for student repos for 545 and 547 ASAP and request the academic thing where you get unlimited private repos for fee.

**Comment by vincenzocoia** [comment]
> and request the academic thing where you get unlimited private repos for free.

Done, and added you @jennybc as an owner to avert limitations.

**Comment by vincenzocoia** [comment]
Reviving this thread, because last year's course structure ended up being too complicated. Here's what I'm suggesting for future years. I'll do the rejigging, along with @firasm. 

- stat545.com contains both the "Topics" tab (@jennybc's realm), and the "current run" of the course (taking the place of last year's [`STAT545-UBC/Classroom`](https://github.com/STAT545-UBC/Classroom) repo / [stat545.com/Classroom](https://stat545.com/Classroom) website). The "Topics" tab is treated like a textbook for the course, and can live here as long as @jennybc wishes. Ultimately, tabs on the stat545.com page will be: Home(=syllabus+people), FAQ, Class Meetings, Assessments, Topics.
- `STAT545-UBC-hw-2019` Org for this year's homework submissions. Old assignment repos living in the `STAT545-UBC` Org are lived with, until I or someone decides to take the time to move them to retroactive `STAT545-UBC-hw-201x` Org(s).
- `STAT545-UBC/Discussion` lives on as usual, where content-related discussion happens.
- `STAT545-UBC-hw-2019/Announcements` will be the course-specific discussion forum (was `STAT545-UBC/Discussion-Internal` last year).

**Comment by jennybc** [comment]
Heads up @vincenzocoia:

Alison Hill (@apreshill) at RStudio and intern Grace Lawley (@gracelawley) have been working on porting STAT 545 content to bookdown over the summer. They have just reached a big milestone and are awaiting input from me. Separately, they have beautification plans. You can see a preview of the basic porting here: https://gracious-allen-b2d17f.netlify.com

One of the motivations is to hand-off a clean, modern version of STAT 545 content to UBC, to evolve as you find useful in the actual course. We will then "ingest" STAT 545 in some more formal way into educational materials hosted by RStudio. Although I have zero immediate plans, you can imagine that the UBC/Vincenzo and RStudio Education/Jenny version will gradually diverge, as they evolve subject to different pressures and constraints.

Basically, we are rapidly approaching our amicable "unconscious uncoupling" in terms of stewarding the STAT 545 content.

So, as you plan for September, you should also plan to lean on content served at a UBC-specific domain or to use the standard one from GitHub Pages. 

Of course, you can still rely on whatever gets hosted within RStudio Education, if you wish, for the slow-changing content ("Topics" tab stuff). As I said, I don't have any grand plans for near-term change and will still welcome issues, PRs, etc. But the source that backs the content served at STAT545.com will soon shift to a different repo belonging to me or an RStudio org. This repo we're talking in can revert to UBC-STAT 545 specific stuff.

Happy to discuss more here!

**Comment by vincenzocoia** [comment]
Ah, that's great that this content is finally being ported to bookdown and hosted by netlify. So, if I'm understanding correctly, both parties will "start" with the same content currently found under the "Topics" tab, and then diverge by both parties modifying the content on our separate mediums (perhaps `rstudio-education` and `STAT545-UBC` Orgs) based on our specific needs? Or does "hand-off a clean, modern version of STAT 545 content to UBC" mean some modified version of the content under the "Topics" tab? 

Also, when do you suspect the hand-off will happen? This would definitely make things a lot cleaner if I can integrate my course materials with the "Topics".

> So, as you plan for September, you should also plan to lean on content served at a UBC-specific domain or to use the standard one from GitHub Pages.

I intend to still use the `STAT545-UBC/STAT545-UBC.github.io` repo to host the course and its corresponding stat545.com site. This format is working well. Or are you hoping to use the stat545.com domain for the bookdown book? Related, will the bookdown book be titled `STAT 545`?

**Comment by jennybc** [comment]
> So, if I'm understanding correctly, both parties will "start" with the same content currently found under the "Topics" tab, and then diverge by both parties modifying the content on our separate mediums

Yep.

> Also, when do you suspect the hand-off will happen?

I suppose you could take a copy of this functioning bookdown site at any point now. Either by replacing the website content here with that or by making a new repo below the STAT545-UBC org that is literally a fork of that. The fork option would allow you to keep pulling changes for a while, while the versions are pretty close.

> Or are you hoping to use the stat545.com domain for the bookdown book? Related, will the bookdown book be titled STAT 545

Yeah, I will use the stat545.com domain for this bookdown book. So yes I will change what stat545.com points to in the not-to-distant future. It will definitely have "STAT 545" on it, but I'm sure we can find a way to describe the history and link to the UBC course in a way tha we're all OK with.

**Comment by vincenzocoia** [comment]
> I suppose you could take a copy of this functioning bookdown site at any point now.

Sounds like a good idea. I might hold off. Going to give it some thought.

> So yes I will change what stat545.com points to in the not-to-distant future.

Okay, I'll see about having UBC host a more permanent domain like stat545-ubc.com, or better yet, stat545.ubc.ca. I'll keep you posted so that you can include a link somewhere on your end.

> but I'm sure we can find a way to describe the history and link to the UBC course in a way tha we're all OK with.

👍

**Comment by vincenzocoia** [comment]
> I'll see about having UBC host a more permanent domain

We now have stat545.stat.ubc.ca. Are you ready to take stat545.com now? Wouldn't want stat545.com to be hanging for very long.

> you can imagine that the UBC/Vincenzo and RStudio Education/Jenny version will gradually diverge, as they evolve subject to different pressures and constraints.

Thinking more about this, I don't imagine STAT 545 @ UBC would _want_ to diverge all that much from the content in RStudio's STAT 545. I imagine RStudio's would be more like a textbook for STAT 545 @ UBC, and UBC would have "worksheet versions" of these notes to be used in lectures (and perhaps its own textbook-style notes if not found in RStudio's stat545.com). Would be interested in hearing more about the stat545.com plans, as well as how you feel about STAT545 @ UBC deliberately moving alongside RStudio's. Also, Tiffany and Paul are interested in hearing more -- would you be up for a quick chat, perhaps over Slack?

**Comment by jennybc** [comment]
@apreshill How close are we to serving stat545.com from something produced from a different repo? Because that frees this repo up for @vincenzocoia et al to make it into what they need for the term starting after Labor Day.

> > you can imagine that the UBC/Vincenzo and RStudio Education/Jenny version will gradually diverge, as they evolve subject to different pressures and constraints.

> Thinking more about this, I don't imagine STAT 545 @ UBC would want to diverge all that much from the content in RStudio's STAT 545. I imagine RStudio's would be more like a textbook for STAT 545 @ UBC, and UBC would have "worksheet versions" of these notes to be used in lectures

Yeah, I agree. There are no plans to do anything that would diverge from this. So we could just regard the impending separation "move" as step 1 in making these things easier to manage separately.

Yeah, I'm happy if you want to create a slack channel or group DM around this in the UBC MDS space.

**Comment by apreshill** [comment]
Hi everyone,

I'm excited we could help make these course materials more portable and accessible!

@jennybc we are ready to serve the stat545.com domain. If you wanted to switch to using Netlify (or perhaps you already have?), we can configure the redirects (see our [current working list of links](https://github.com/gracelawley/stat545-rstudio/blob/time-travel/time-travel/reboot-ubc-ref-mappings.csv)); otherwise, we can look into the redirect rules for your current hosting platform.

**Comment by vincenzocoia** [comment]
Success! We now have:

- https://stat545.stat.ubc.ca/ as the main STAT 545 @ UBC website
    - corresponding repo [`STAT545-UBC/STAT545-home`](https://github.com/STAT545-UBC/STAT545-home)
- https://stat545guidebook.netlify.com as the "lecture notes"
    - corresponding (repurposed) repo [`STAT545-UBC/Classroom`](https://github.com/STAT545-UBC/Classroom)
    - may host this at https://stat545.stat.ubc.ca/guidebook down the road

That means that this very repo and its site will probably become obsolete, to be re-directed to the main site.

**Comment by vincenzocoia** [comment]
> to be re-directed to the main site.

Actually, scrap that, since stat545.com will already be pointing elsewhere.

**Comment by jennybc** [comment]
Congrats! OK this has lit a match under me re: turning the other site on, i.e. serving what @apreshill and @gracelawley have created via stat545.com. More soon.

**Comment by jennybc** [comment]
Hi @vincenzocoia I think this can happen now:

I point use stat545.com domain for this content: <https://stat545-book.netlify.com>

Q1: Are you OK if that happens in the next ~24 hours? I plan to keep refining what's there, but we've crossed the threshold where it's clearly better than the current arrangements.

Q2: When I link back to the current UBC course, I'm linking to <https://stat545.stat.ubc.ca>. Please confirm that's correct.

Q3: Shall we rename this repo to something that indicates its history, but gets it sort of "out of your way"? I'd like it to still exist and to be public, for historical sake and there could be useful info here. Possible name: STAT-545-UBC-old-website.

**Comment by jennybc** [comment]
Also, @gracelawley has done a great job with redirects so that existing links to STAT545 content should redirect to the correct sub-pages, but with less annoying names. Example:

Original page: <http://stat545.com/block010_dplyr-end-single-table.html>

Will redirect to: `https://stat545.com/dplyr-single.html` after the transition

We can sort of preview / test this functionality by visiting

<https://stat545-book.netlify.com/block010_dplyr-end-single-table.html>

and confirming that it goes to

<https://stat545-book.netlify.com/dplyr-single.html>

Redirect rules don't include the domain/hostname, so I feel confident this will continue to work after I point the domain at the Netlify-hosted content.

**Comment by vincenzocoia** [comment]
Q1: Yep, happy for this transition to happen anytime.

Q2: The URL https://stat545.stat.ubc.ca/ is indeed the correct pointer for STAT 545 @ UBC going forward.

Q3: I don't mind this repo keeping its name, and that might be the least confusing option. But I'm not opposed to a name change. 

And thanks for letting me know about the redirects, that should be useful. Either way, I'll try to be vigilant about using the newest URL's.

**Comment by jennybc** [comment]
OK I've changed which content stat545.com points to. It's basically / sort of working. I think the current glitch re: https / certificate will self-resolve or I'll be able to resolve over the next 24 hours.

I think I would like this repo's name to change, just for clarity. Additionally, I propose to edit the README explaining that it's "frozen". We could then [archive it formally](https://help.github.com/en/articles/archiving-repositories).

**Comment by jennybc** [comment]
I now think that changing the name of this repo might also be favourable for the web transition:

https://help.github.com/en/articles/unpublishing-a-user-pages-site

I would very much like to unpublish this site. We may eventually get to the need to delete `master` branch or delete the repo. But I think renaming is something we should do anyway and might help in my efforts to get Netlify serving an https site.

**Comment by jennybc** [comment]
Renamed to `STAT545-UBC-original-website`. Good for humans and, I hope, for computers and internet things.

**Comment by gracelawley** [comment]
Hi @jennybc I didn't realize that you wanted to unpublish the original site when I made the redirects. I thought that (what is now) the `STAT545-UBC-original-website` repo would continue to exist at <https://STAT545-UBC.github.io>, so I had any pages that didn't make it into the bookdown reboot redirect to the gh pages site in an effort to save the rendered content. For example,

Original page: <https://stat545.com/111Scraping_Workthrough.html>

Redirects to: <https://STAT545-UBC.github.io/STAT545-UBC.github.io/111Scraping_Workthrough.html>

There are 31 urls that currently redirect to the gh pages site (I've attached a txt of them) - not all of these appear in the bookdown reboot but I know that at least 9 of them are in the Deprecated section of the appendix (see [here](https://stat545.com/oldies.html)). 

I could make these 31 redirect to the raw `Rmd` instead, so for example,
[ghpages-redirects.txt](https://github.com/STAT545-UBC/STAT545-UBC-original-website/files/3619278/ghpages-redirects.txt)

Original page: <https://stat545.com/111Scraping_Workthrough.html>

Redirects to: <https://github.com/STAT545-UBC/STAT545-UBC-original-website/blob/3c8be1358f28af871012f18b1b37db3b52351d4d/111Scraping_Workthrough.Rmd>

What do you think?

**Comment by jennybc** [comment]
I think redirecting to the Rmd, in the original website repo, is a good option of last resort.

Slightly better options that we might be able to work out:

  * Deciding that the content is best retired or link elsewhere.
  * Linking to something maintained below the current STAT 545 UBC course's site.

@vincenzocoia please weigh in if you have an opinion or are planning to maintain an exact copy of the old site for a while. Otherwise, we'll fall back to linking too `.Rmd` and/or retiring.

**Comment by vincenzocoia** [comment]
> I propose to edit the README explaining that it's "frozen". We could then archive it formally.

> I would very much like to unpublish this site.

Sure. Perhaps when we close this Issue.

Also, I don't intend to refer to anything that doesn't end up in the STAT 545 book, so retiring the content in this repository isn't a big deal as far as STAT 545 @ UBC is concerned. It might be useful if these old tutorials like `ddply()` remain available, but probably not worth the effort of publishing on a website like STAT545-UBC.github.io.

**Comment by jennybc** [comment]
I think we should just declare that content "retired". And if we need to link to it, we should link to **rendered markdown** (not the raw `.Rmd`) here.

@gracelawley The reason we don't want to maintain some legacy website is that, even if the content isn't changing, it still requires some care and feeding. And it's just not worth it to any of us. I also think it would be confusing. We've already got to explain the difference between stat545.com and <https://stat545.stat.ubc.ca/>.

**Comment by jennybc** [comment]
Closing this so I can archive the repo. We can undo this if necessary, but it seems pretty safe to do now.
